web/packages/extension/src/ruffle.ts

```diff
@@ -57,6 +58,60 @@ function openInNewTab(swf: URL): void {
     window.postMessage(message, "*");
 }
 
+function polyfillCurrentScript(): void {
+    const script = document.currentScript!;
+    try {
+        if (determineKnownTypeHasNoEmptyStringProp(script, "src")) {
+            determineScriptIsMaskedSrcAndMakePolyfill(script);
+        }
+    } catch (_) {
+        // Continue to run
+    }
+}
+
+function determineScriptIsMaskedSrcAndMakePolyfill(
+    script: HTMLScriptElement,
+): void {
+    const scriptUrlPolyfill = script.getAttribute("ruffle-src-polyfill");
+    if (!scriptUrlPolyfill) {
+        return;
+    }
+    const scriptUrl = script.src;
```

Comment:
Starting here, up to the end of the function, I think we can just replace this with something much simpler:

```js
    const scriptAutoPublicPath = getWebpackPublicPathFromScriptSrc(scriptUrl);
    const polyfillPath = getWebpackPublicPathFromScriptSrc(scriptUrlPolyfill);

    if (__webpack_public_path__ === scriptAutoPublicPath) {
        // TODO: If there are other scripts that need to be dynamically created and executed, the current polyfill logic should also be applied.
        __webpack_public_path__ = polyfillPath;
    }
    // TODO: Process other situations when mask url not webkit-masked-url://hidden/
```